src/uc_paymentwall.module

```diff
@@ -2,308 +2,325 @@
+    );
 
-  return $form;
+    return $form;
 }
 
 /**
- * @param $data
- * Init Paymentwall config
+ * Complete order on example.com/ubercart_install_folder/paymentwall_complete
  */
-function init_paymentwall_config($data){
-  Paymentwall_Config::getInstance()->set(array(
-      'api_type' => Paymentwall_Config::API_GOODS,
-      'public_key' => $data->app_key, // available in your Paymentwall merchant area
-      'private_key' => $data->secret_key // available in your Paymentwall merchant area
-  ));
+function uc_paymentwall_complete()
+{
+    switch ($_SERVER['REQUEST_METHOD']) {
+        case 'GET':
+            drupal_goto(); // Redirect to home page
+            die();
+            break;
+
+        case 'POST':
+            global $user;
+
+            if (isset($_SESSION['order'])) {
+
+                initPaymentwallConfigs();
+                $products_name = array();
+                $orderData = unserialize(base64_decode($_SESSION['order']));
+
+                // Create new Order
+                $order = uc_order_new($user->uid);
+                $orderData->order_id = $order->order_id;
+
+                // Save order data
+                uc_order_save($orderData);
+
+                foreach ($orderData->products as $key => $value) {
+                    if (!in_array($value->title, $products_name)) {
+                        array_push($products_name, $value->title);
+                    }
+                }
+
+                $widget = new Paymentwall_Widget(
+                    $orderData->uid ? $orderData->uid : $orderData->primary_email,
+                    variable_get('uc_pw_widget_code'),
+                    array(
+                        new Paymentwall_Product(
+                            $orderData->order_id,
+                            number_format($orderData->order_total, 2, '.', ''),
+                            $orderData->currency,
+                            implode(', ', $products_name),
+                            Paymentwall_Product::TYPE_FIXED
+                        )
+                    ),
+                    array_merge(
+                        array(
+                            'email' => $orderData->primary_email,
+                            'integration_module' => 'ubercart',
+                            'success_url' => variable_get('uc_pw_success_url'),
+                            'test_mode' => variable_get('uc_pw_test_mode'),
+                        ),
+                        getUserProfileData($orderData)
+                    )
+                );
+
+                // Clear cart
+                uc_cart_empty(uc_cart_get_id());
+                unset($_SESSION['order']);
+
+                return $widget->getHtmlCode();
+            } else {
+                drupal_goto();
+                die();
+            }
+
+            break;
+    }
 }
 
 /**
- * Complate order on example.com/paymentwall_complete
+ * Ping back example.com/ubercart_install_folder/paymentwall_pingback
```

Comment:
still incorrect, because ubercart can be installed to example.com

let's use abstract 
`{path_to_ubercart}/paymentwall_pingback`